elastic/kibana
# [Dashboard] Change panel filters action visibility

Closes https://github.com/elastic/kibana/issues/141536
Closes https://github.com/elastic/kibana/issues/143386

## Summary

https://github.com/elastic/kibana/pull/136655 introduced a panel level filters badge that would notify users when a visualization was being filtered/queried on an individual panel level. While this exposed important information about how the panels were being generated and made it so that users no longer had to navigate to Lens/Visualize/etc. to see this information, it introduced a lot of undesired clutter to dashboards. This was especially annoying for dashboards in **view** mode, since panel level filter/query information is primarily useful for dashboard authors and thus this badge had minimal benefit despite its large visual impact.

To fix this, this PR does three things:
1. Converts the panel filters badge to a single notification icon
    
    ![BeforeAndAfterFilterBadge](https://user-images.githubusercontent.com/8698078/205696453-c8a7bc85-46bf-4a0c-b206-bbb690577d9a.png)

2. Displays the filters in a popover rather than a modal


    ![BeforeAndAfterModal](https://user-images.githubusercontent.com/8698078/205758567-314e841d-646e-4cb6-973d-0fde79c2b9df.png)


3. Makes this notification icon only visible in **edit** mode - it no longer shows up in view or print mode.

### How to Test
1. Create a visualization with panel level filters and/or queries.
    - Note that the sample `[Flights] Global Flight Dashboard` dashboard already has some visualizations with panel level filters, so you could also use that instead.
3. Ensure that the panel level filters action icon **only** shows up in edit mode
    - Swap to view mode and make sure there is no icon
    - Print a PDF or PNG report and make sure there is no icon
4. Ensure that the `"Edit filters"` button in the popover keeps context when clicked, including allowing you to `"Save and return"` after being taken to Lens/Visualize/etc.
5. Ensure that the contents of the popover react to changes to panel level filters/queries

### Checklist




### For maintainers
